Framework/Algorithms/src/WorkspaceJoiners.cpp

```diff
@@ -56,19 +56,14 @@ WorkspaceJoiners::execWS2D(API::MatrixWorkspace_const_sptr ws1,
     auto &outSpec = output->getSpectrum(i);
     const auto &inSpec = ws1->getSpectrum(i);
 
-    // Copy X,Y,E
-    outSpec.setX(XValues);
-    outSpec.dataY() = inSpec.dataY();
-    outSpec.dataE() = inSpec.dataE();
+    outSpec.setHistogram(inSpec.histogram());
```

Comment:
I ran a performance measurement, and I see improvement in the range of 30-60%, depending on workspace size and iterations, tested on Mantid 3.7 versus current master release build, however my performance tests have been inaccurate before, so if you'd like to check/confirm the performance difference for yourself I'm leaving my testing script: https://gist.github.com/DimitarTasev/b02979ae42743fcf520eb00c76e6f18a